File: product-service/src/main/java/com/intexsoft/webshop/productservice/service/ProductService.java
package com.intexsoft.webshop.productservice.service;

import com.intexsoft.webshop.productservice.dto.product.ProductCreateDto;
import com.intexsoft.webshop.productservice.dto.product.ProductDto;
import com.intexsoft.webshop.productservice.dto.product.ProductUpdateDto;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface ProductService {
    ProductDto createProduct(ProductCreateDto productCreateDto);

    ProductDto findProductById(Long productId);

    List<ProductDto> findProducts(Pageable pageable);

    ProductDto updateProduct(Long productId, ProductUpdateDto productUpdateDto);

    void deleteProductById(Long productId);
}

File: product-service-kt/src/main/kotlin/com/intexsoft/webshop/productservicekt/service/ProductService.kt
package com.intexsoft.webshop.productservicekt.service

import com.intexsoft.webshop.productservicekt.dto.product.ProductCreateDto
import com.intexsoft.webshop.productservicekt.dto.product.ProductDto
import com.intexsoft.webshop.productservicekt.dto.product.ProductUpdateDto
import org.springframework.data.domain.Pageable

interface ProductService {
    fun createProduct(productCreateDto: ProductCreateDto): ProductDto
    fun findProductById(productId: Long): ProductDto
    fun findProducts(pageable: Pageable): List<ProductDto>
    fun updateProduct(productId: Long, productUpdateDto: ProductUpdateDto): ProductDto
    fun deleteProductById(productId: Long)
}